src/pages/components/RevokeActionsButton.js

```diff
@@ -1,37 +1,214 @@
+/* eslint-disable prettier/prettier */
 import React from 'react'
 import styled, { keyframes } from 'styled-components'
 import { Button } from '@digicatapult/ui-component-library'
 
-const reasonsDummyJSON = {
-  predefinedReasons: {
-    0: { selection: null },
-    1: { selection: null },
-    2: { selection: null },
-  },
+import { useState, useCallback, useMemo } from 'react'
+import { useRef } from 'react'
+import { Dialog } from '@digicatapult/ui-component-library'
+
+import { Section } from '@digicatapult/ui-component-library'
+import WarningSignSvg from '../../assets/images/warning-sign-icon.svg'
+
+// import { useEffect } from 'react'
+
+const reasonsDefaultObj = {
+  dataError: { discrepancies: false, incorrect: false, missing: false },
+  certMis: { incorrect: false, unverified: false, claims: false },
+  nonCompliance: { violation: false, repeated: false },
   otherReason: '',
 }
 
+const ReasonsPopup = ({ handleConfirm }) => {
+  const [isConfVisible, setIsConfVisible] = useState(false)
+  const [reasons, setReasons] = useState(reasonsDefaultObj)
+  // const [text, setText] = useState('')
+  const isValid = useMemo(() => {
+    const hasAtLeastOneItem = Object.values(reasons)
+      .flatMap((o) => Object.values(o))
+      .some((v) => v === true)
+    const hasOtherReason = reasons.otherReason.trim().length > 0
+    return hasAtLeastOneItem || hasOtherReason
+  }, [reasons])
+  const update = (group, pair) => {
+    setReasons({
+      ...reasons,
+      [group]: {
+        ...reasons[group],
+        [pair]: !reasons[group][pair],
+      },
+    })
+  }
+  const change = (text) => {
+    setReasons({ ...reasons, otherReason: text })
+  }
+  return (
+    <>
+      {!isConfVisible && (
+        <>
+          <Section
+            headingLevel={1}
+            title=""
+            background="#27847a"
+            headingSize="0em"
+          >
+            <DivWarning>
+              <TitleWarning>
+                Please provide the reason: <br />
+                <SubtitleWarning>
+                  Submit button to cause revocation in the certification that
```

Comment:
We shouldn't mention 'blockchain' anywhere. Please change this to the simpler: 
"Click the submit button to revoke the certificate"

This also means we need an obvious way to cancel certification - can we have a cancel button next to the submit button, or a big `X` in the top right of the dialog?

Reply:
<img width="985" alt="image" src="https://github.com/digicatapult/sqnc-hyproof-client/assets/58742260/09a3adb8-96e2-404a-9aec-8bf7ef2976c2">

Reply:
Done